src/CodeGen_OpenCL_Dev.cpp

```diff
@@ -16,6 +17,131 @@ using std::sort;
 
 static ostringstream nil;
 
+// OpenCL doesn't support vectors of bools, this mutator rewrites IR
+// to use signed integer vectors instead.
+class EliminateBoolVectors : public IRMutator {
+private:
+    using IRMutator::visit;
+
+    template <typename T>
+    void visit_comparison(const T* op) {
+        Expr a = mutate(op->a);
+        Expr b = mutate(op->b);
+
+        Type t = a.type();
+        if (t.width > 1) {
+            // To represent bool vectors, OpenCL uses vectors of signed
+            // integers with the same width as the types being compared.
+            t.code = Type::Int;
+            expr = T::make(a, b);
```

Comment:
Sorry I have not had time to try and craft a test case. Here is a sketch to consider however:
    Param<bool> test;
    ImageParam foo(...);
    ... select((foo(x, y) < 1) == test), ...) ...

(Scheduled with vectorization of course.)

In this case I would expect the inner comparison to be promoted to an integer of width > 1, but not the Param, which results in badly formed IR.